src/DataSet/ObjectDataSet.php

```diff
@@ -64,25 +71,26 @@ public function getRules(): iterable
         }
 
         if ($this->hasCacheItem('rules')) {
-            return $this->getCacheItem('rules');
-        }
-
-        $rules = [];
-        foreach ($this->getReflectionProperties() as $property) {
-            // TODO: use Generator to collect attributes.
-            $attributes = $property->getAttributes(RuleInterface::class, ReflectionAttribute::IS_INSTANCEOF);
-            foreach ($attributes as $attribute) {
-                $rule = $attribute->newInstance();
-                $rules[$property->getName()][] = $rule;
-
-                if ($rule instanceof AttributeEventInterface) {
-                    $rule->afterInitAttribute($this);
+            /** @var iterable $rules */
+            $rules = $this->getCacheItem('rules');
+        } else {
```

Comment:
Was it necessary to change early return to `else`?

Reply:
Yes, it was. Otherwise StyleCi will remove "unnecessary" variable `$rules` (PHPDoc will still remain though) resulting to this code:

```php
 /** @var iterable $rules */
return $this->getCacheItem('rules');
```

But it's needed for declaring explicit type via PHPDoc, otherwise Psalm won't be able to recognize it. I would prefer early return here too, but I couldn't find an other way to fix it.

Reply:
Try

```php
/** @var iterable */
return $this->getCacheItem('rules');
```

Better suppress error with comment instead remove early return.

Reply:
Actually I haven't used PHPDoc comments for returned value directly (without separate variable) before. I checked it - it's valid and works fine, so let's fix it that way. Thanks.